atomic_reactor/plugins/pre_cancel_builds_on_autorebuild.py

```diff
@@ -0,0 +1,83 @@
+"""
+Copyright (c) 2017 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+
+from osbs.api import OSBS
+from osbs.conf import Configuration
+from osbs.exceptions import OsbsResponseException
+from atomic_reactor.util import get_build_json
+from atomic_reactor.plugin import PreBuildPlugin
+from atomic_reactor.plugins.pre_check_and_set_rebuild import is_rebuild
+
+
+class CancelBuildsOnAutoRebuild(PreBuildPlugin):
+    """
+    If the current build is rebuild, we need to query OSBS to make sure there
+    are no other builds currently running for this container and build target
+    """
+    key = 'cancel_builds_on_autorebuild'
+
+    is_allowed_to_fail = False
+
+    def __init__(self, tasker, workflow, url, verify_ssl=True, use_auth=True):
+        """
+        constructor
+
+        :param tasker: DockerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        :param url: str, URL to OSv3 instance
+        :param verify_ssl: bool, verify SSL certificate?
+        :param use_auth: bool, initiate authentication with OSv3?
+        """
+        super(CancelBuildsOnAutoRebuild, self).__init__(tasker, workflow)
+
+        self.url = url
+        self.verify_ssl = verify_ssl
+        self.use_auth = use_auth
+
+    def run(self):
+
+        metadata = get_build_json().get("metadata", {})
+        labels = metadata.get("labels", {})
+        buildconfig = labels["buildconfig"]
+
+        self.log.info("This is a rebuild? %s", buildconfig)
+        if not is_rebuild(self.workflow):
+            self.log.info(
+                'this is not an autorebuild, %s is doing nothing' % self.key
```

Comment:
Because this is a logging method, the preferred way is `self.log.info('... %s ...', self.key)`.